kyo-core/shared/src/main/scala/kyo/Duration.scala

```diff
@@ -0,0 +1,74 @@
+package kyo
+
+import java.time.Duration as JDuration
+import java.time.temporal.ChronoUnit.*
+import java.time.temporal.TemporalUnit
+import scala.annotation.switch
+import scala.concurrent.duration.Duration as SDuration
+
+opaque type Duration <: JDuration = JDuration
+given CanEqual[Duration, Duration] = CanEqual.derived
```

Comment:
Can we define this manually?

Reply:
You'd just to implement an instance of `CanEqual` here I think. Something important about Scala 3's encoding of `given` is that it introduces lazy initialization under the hood, which requires a memory barrier on each read and prevents JIT optimizations in some cases. I haven't migrated the codebase yet but this is the encoding I've been using to avoid the overhead:

```
class SomeType

private val someValue = new SomeType
inline given SomeType = someValue
```